drivers/sx126x/sx126x_netdev.c

```diff
@@ -59,15 +59,18 @@ static int _send(netdev_t *netdev, const iolist_t *iolist)
     }
 
     DEBUG("[sx126x] netdev: sending packet now (size: %d).\n", size);
+    int pos = 0;
     /* Write payload buffer */
     for (const iolist_t *iol = iolist; iol; iol = iol->iol_next) {
         if (iol->iol_len > 0) {
-            sx126x_set_lora_payload_length(dev, iol->iol_len);
-            sx126x_write_buffer(dev, 0, iol->iol_base, iol->iol_len);
+            sx126x_write_buffer(dev, pos, iol->iol_base, iol->iol_len);
             DEBUG("[sx126x] netdev: send: wrote data to payload buffer.\n");
+            pos += iol_len;
         }
     }
 
+    sx126x_set_lora_payload_length(dev, size);
```

Comment:
Could it be pos instead of size ?

Reply:
why?

If `size` != `pos`, the `iolist_size` implementation would be wrong. I would even add an assert there

Reply:
(I have no problem to change it, but just wanted to know why :P)

Reply:
ok, I adapted the logic a bit to avoid traversing 2 iolists. Now I don't use `iolist_size` anymore.

Reply:
I'm fine with the changes but can't test them. It would be nice to have testing result for gnrc_lorawan but also for drivers_sx126x and semtech-loramac package.

Reply:
unfortunately we only have a `nucleo-wljc55` whose radio is still not finished :/
@fjmolinas do you  have by chance an external `sx126x` or similar?